src/prompt/interaction.rs

```diff
@@ -86,50 +92,56 @@ pub trait PromptInteraction<T> {
                 prev_frame = frame;
             }
 
-            if let State::Submit(result) = state {
-                return Ok(result);
-            }
-
-            if let State::Cancel = state {
-                return Err(io::ErrorKind::Interrupted.into());
+            match state {
+                State::Submit(result) => return Ok(result),
+                State::Cancel => return Err(io::ErrorKind::Interrupted.into()),
+                _ => {}
             }
 
-            let key = term.read_key()?;
-
-            if let Some(cursor) = self.input() {
-                match key {
-                    Key::Char(chr) if !chr.is_ascii_control() => {
-                        cursor.insert(chr);
-                    }
-                    Key::Backspace => {
-                        cursor.delete_left();
-                    }
-                    Key::Del => {
-                        cursor.delete_right();
-                    }
-                    Key::ArrowLeft => {
-                        cursor.move_left();
+            match term.read_key() {
+                Ok(Key::Escape) => state = State::Cancel,
+
+                Ok(key) => {
+                    let word_editing = self.allow_word_editing();
+                    if let Some(cursor) = self.input() {
+                        match key {
+                            Key::Char(chr) if !chr.is_ascii_control() => cursor.insert(chr),
+                            Key::Backspace => cursor.delete_left(),
+                            Key::Del => cursor.delete_right(),
+                            Key::ArrowLeft => cursor.move_left(),
+                            Key::ArrowRight => cursor.move_right(),
+                            Key::Home => cursor.move_home(),
+                            Key::End => cursor.move_end(),
+
+                            // Alt-Backspace
+                            Key::Char('\u{17}') if word_editing => cursor.delete_word_left(),
+
+                            // Alt-ArrowLeft
```

Comment:
Basically, `Ctrl-Left`/`Ctrl-Right` behave similarly in Bash, could `Ctrl` be enabled as well? Thanks!

Reply:
yes... but i propose doing so in a separate PR :)